pennylane_lightning_gpu/src/tests/mpi/Test_MPI_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -296,3012 +296,221 @@ TEMPLATE_TEST_CASE("StateVectorCudaMPI::GateOpsNonParam",
-                          {0.17324175995006008, 0.12834320562185453},
-                          {-0.021542232643170886, -0.2537776554975786},
-                          {0.2917899745322105, 0.30227665008366594},
-                          {-0.17082687702494623, -0.013880922806771745},
-                          {0.03801974084659355, 0.2233816291263903},
-                          {-0.1991010562067874, -0.2378546697582974},
-                          {0.13833362414043807, 0.0571737109901294},
-                          {-0.1960850292216881, -0.22946370987301284}}};
-
-    auto local_state = mpi_manager.scatter(init_sv, 0);
-    auto expected_sv0 = mpi_manager.scatter(expected_sv[0], 0);
-    auto expected_sv1 = mpi_manager.scatter(expected_sv[1], 0);
-    mpi_manager.Barrier();
-
-    int nDevices = 0; // Number of GPU devices per node
-    cudaGetDeviceCount(&nDevices);
-    int deviceId = mpi_manager.getRank() % nDevices;
-    cudaSetDevice(deviceId);
-    mpi_manager.Barrier();
+    StateVectorCudaMPI<TestType> sv(mpi_manager, nGlobalIndexBits,
+                                    nLocalIndexBits);
+    sv.CopyHostDataToGpu(local_state, false);
+    PrecisionT mpi_result, result;
+    std::vector<size_t> wire{0};
 
-    SECTION("Apply directly") {
-        SECTION("Operation on globalQubits") {
-            StateVectorCudaMPI<TestType> sv(mpi_manager, nGlobalIndexBits,
-                                            nLocalIndexBits);
-            sv.CopyHostDataToGpu(local_state, false);
-            sv.applyPauliZ({0}, false);
-            sv.CopyGpuDataToHost(local_state.data(),
-                                 static_cast<std::size_t>(subSvLength));
-            CHECK(local_state == Pennylane::approx(expected_sv0));
-        }
+    std::vector<cp_t> matrix(4, cp_t(1.0, 0.0));
 
-        SECTION("Operation on localQubits") {
-            StateVectorCudaMPI<TestType> sv(mpi_manager, nGlobalIndexBits,
-                                            nLocalIndexBits);
-            sv.CopyHostDataToGpu(local_state, false);
-            sv.applyPauliZ({num_qubits - 1}, false);
-            sv.CopyGpuDataToHost(local_state.data(),
-                                 static_cast<std::size_t>(subSvLength));
-            CHECK(local_state == Pennylane::approx(expected_sv1));
-        }
-    }
+    matrix[1] = cp_t(0, 0);
+    matrix[2] = cp_t(0, 0);
 
-    SECTION("Apply using dispatcher") {
-        SECTION("Operation on globalQubits") {
-            StateVectorCudaMPI<TestType> sv(mpi_manager, nGlobalIndexBits,
-                                            nLocalIndexBits);
-            sv.CopyHostDataToGpu(local_state, false);
-            sv.applyOperation("PauliZ", {0}, false);
-            sv.CopyGpuDataToHost(local_state.data(),
-                                 static_cast<std::size_t>(subSvLength));
-            CHECK(local_state == Pennylane::approx(expected_sv0));
-        }
+    mpi_result = sv.expval(wire, matrix).x;
```

Comment:
Do we test this for all possible matrix values? As in, all expected gates and a number of variations?

Reply:
Let me add more tests.